synapse/rest/admin/rooms.py

```diff
@@ -231,6 +231,30 @@ async def on_GET(self, request, room_id):
         return 200, ret
 
 
+class RoomMembersRestServlet(RestServlet):
+    """
+    Get members list of a room.
+    """
+
+    PATTERNS = admin_patterns("/rooms/(?P<room_id>[^/]+)/members")
+
+    def __init__(self, hs):
+        self.hs = hs
+        self.auth = hs.get_auth()
+        self.store = hs.get_datastore()
+
+    async def on_GET(self, request, room_id):
+        await assert_requester_is_admin(self.auth, request)
+
+        members = await self.store.get_users_in_room(room_id)
+        if not members:
+            raise NotFoundError("Room not found or empty")
```

Comment:
After playing with this a bit more, I think this is going to be confusing, I think we should add use the `self.store.get_room(room_id)` method first, if that returns `None` we can raise `NotFoundError`.

Then, if there are no members I think we should return it as a success: `{"members": [], "total": 0}`.

Reply:
Thats a very good suggestion. I'll add the check.